prov/ucx: base address with FI_MR_VIRT_ADDR and FI_MR_RAW

**Describe the bug**
The UCX provider requires FI_MR_RAW for memory registrations; therefore, we must call `fi_mr_raw_attr` on the MR to obtain the raw key and base address, as per the documentation here: https://ofiwg.github.io/libfabric/v2.3.1/man/fi_mr.3.html. The base address returned from `fi_mr_raw_attr` with ucx is zero. 
When using this address (zero) for doing a remote read, the code fails with 

```
ib_mlx5_log.c:179  Remote access error on mlx5_0:1/IB (synd 0x13 vend 0x88 hw_synd 0/0)
ib_mlx5_log.c:179  RC QP 0x95a4 wqe[0]: RDMA_READ s-- [rva 0x0 rkey 0x2e40f5] [va 0x7f3300d0b540 len 32768 lkey 0x2e20c5] [rqpn 0x95a5 dlid=1 sl=0 port=1 src_path_bits=0]
```

Using the virtual address as the base address works, and the code passes through. 

**To Reproduce**
register a buffer for remote read, use ucx provider, use  `FI_MR_VIRT_ADDR` and `FI_MR_RAW` flags. use the returned base address from  `fi_mr_raw_attr` to do the remote read RMA call.

**Expected behavior**
`fi_mr_raw_attr` should return a valid address to use in the rma call.

**Output**
```
ib_mlx5_log.c:179  Remote access error on mlx5_0:1/IB (synd 0x13 vend 0x88 hw_synd 0/0)
ib_mlx5_log.c:179  RC QP 0x95a4 wqe[0]: RDMA_READ s-- [rva 0x0 rkey 0x2e40f5] [va 0x7f3300d0b540 len 32768 lkey 0x2e20c5] [rqpn 0x95a5 dlid=1 sl=0 port=1 src_path_bits=0]
```

@j-xiong That was quick. I confirm the fix is working fine for me. Thanks for handling this. 

(minor) I get this warning about a missing cast here: 

```
 prov/ucx/src/ucx_domain.c:289:20: warning: assignment to ‘uint64_t’ {aka ‘long unsigned int’} from ‘void * const’ makes integer from pointer without a cast [-Wint-conversion]

 289 |  ucx_mr->base_addr = attr->mr_iov->iov_base;
```

--- a/prov/ucx/src/ucx.h
+++ b/prov/ucx/src/ucx.h
@@ -113,6 +113,7 @@ struct ucx_fabric {
 struct ucx_mr {
 	struct ofi_mr omr;
 	ucp_mem_h memh;
+	uint64_t base_addr;
 };
 
 #define FI_UCX_PKEY_SIGNATURE	(0x20230116ULL)

--- a/prov/ucx/src/ucx_domain.c
+++ b/prov/ucx/src/ucx_domain.c
@@ -185,8 +185,8 @@ static int ucx_mr_control(struct fid *fid, int command, void *arg)
 	memcpy(raw_attr->raw_key, tmp, tmp_size);
 	*raw_attr->key_size = tmp_size;
 	ucp_rkey_buffer_release(tmp);
+	*raw_attr->base_addr = ucx_mr->base_addr;
 
-	/* TODO: should (*raw_attr->base_addr) be set? unused for now. */
 	return FI_SUCCESS;
 }
 
@@ -286,6 +286,7 @@ static int ucx_mr_regattr(struct fid *fid, const struct fi_mr_attr *attr,
 		goto out;
 	}
 
+	ucx_mr->base_addr = (uint64_t)(uintptr_t) attr->mr_iov->iov_base;
 	mr->mr_fid.key = mr->key = key;
 	mr->mr_fid.mem_desc = (void *)(uintptr_t) key;
 

